packages/angular/projects/ui-angular/src/lib/primitives-poc/text/text-component/text.component.ts

```diff
@@ -14,14 +14,15 @@ const $COMPONENT_SELECTOR: string = '[amplify-text-poc]';
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
+
+/** Base primitive for property binding */
 export class AmplifyTextComponent extends AmplifyBasePrimitiveComponent {
   /** Input Prop for style changes  */
 
-  @Input() variation: boolean = false;
+  @Input() variation: string = 'primary';
```

Comment:
Can we strictly type this?

```ts
type TextVariation = 'primary' | 'secondary' | 'tertiary' | 'error' | 'warning' | 'info' | 'success'
```

Reply:
https://github.com/aws-amplify/amplify-ui/pull/2415/commits/f3d9e3121d464ab6cdf223cbf3f9967bce3ac4dc